src/app/qgsrelationadddlg.cpp

```diff
@@ -24,45 +25,41 @@ QgsRelationAddDlg::QgsRelationAddDlg( QWidget *parent )
 {
   setupUi( this );
 
-  mTxtRelationId->setPlaceholderText( tr( "[Generated automatically]" ) );
-  checkDefinitionValid();
+  connect( mCbxReferencingLayer, &QgsMapLayerComboBox::layerChanged, this, [=]( QgsMapLayer * layer ) { mCbxReferencingField->setLayer( layer ); }
```

Comment:
Seems not supported in the embedded astyle version, but supported in more recent versions. How's #3463 going?